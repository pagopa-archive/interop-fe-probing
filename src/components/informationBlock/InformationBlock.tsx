import { Chip, Grid, Typography, Button } from '@mui/material'
import { InformationContainer } from '@pagopa/interop-fe-commons'
import LaunchIcon from '@mui/icons-material/Launch'
import LockIcon from '@mui/icons-material/Lock'
import RefreshIcon from '@mui/icons-material/Refresh'

interface IService {
  eService: string
  regulator: string
  version: string
  statusMonitoring: string
  statusService: string
  statusLastDetection: string
}

interface IProps {
  data: IService
  reloadInfoBlock?: () => void
  viewInCatalogue?: () => void
}

enum Labels {
  eService = 'SCHEDA E-SERVICE',
  regulator = 'EROGATORE',
  version = 'VERSIONE',
  statusMonitoring = 'STATO DEL MONITORAGGIO',
  statusService = "STATO DEL'E SERVICE",
  statusLastDetection = 'STATO ULTIMA RILEVAZIONE',
}

const blocksInfo: Array<{ title: string; blocks: Array<string> }> = [
  {
    title: 'Informazioni generali',
    blocks: ['regulator', 'version', 'eService'],
  },
  {
    title: 'In tempo reale',
    blocks: ['statusMonitoring', 'statusService', 'statusLastDetection'],
  },
]

export const InformationBlock: React.FC<IProps> = ({ data, reloadInfoBlock, viewInCatalogue }) => {
  return (
    <>
      <Grid container direction={'column'} alignItems="center" rowGap={5} mb={5}>
        {blocksInfo.map((blockInfo) => (
          <Grid item container direction="column" key={blockInfo.title}>
            {blockInfo.title === 'In tempo reale' && (
              <>
                <Grid item textAlign={'center'}>
                  <Typography
                    sx={{
                      fontSize: '1.4em',
                      fontWeight: 'bold',
                      color: '#17324D',
                    }}
                  >
                    {blockInfo.title}
                  </Typography>
                </Grid>

                <Grid
                  item
                  container
                  xs={6}
                  px={2}
                  my={2}
                  justifyContent={'end'}
                  sx={{ backgroundColor: '#F2F2F2' }}
                >
                  <Button
                    onClick={reloadInfoBlock}
                    color={'primary'}
                    disableRipple
                    sx={{
                      fontWeight: 'bold',
                      border: 'none!important',
                      textTransform: 'none',
                      '&:hover': {
                        backgroundColor: 'transparent',
                      },
                    }}
                    variant="outlined"
                    startIcon={<RefreshIcon />}
                  >
                    Ricarica
                  </Button>
                </Grid>
              </>
            )}
            <Grid item container direction={'column'}>
              {blockInfo.blocks.map((block) => {
                return block === 'eService' ? (
                  <Grid item key={block} mt={2}>
                    <InformationContainer
                      key={block}
                      label={Labels[block as keyof typeof Labels]}
                      content={
                        <Grid container>
                          <Button
                            color={'primary'}
                            disableRipple // remove onClick effect
                            sx={{
                              fontWeight: 'bold',
                              border: 'none!important',
                              textTransform: 'none',
                              p: '0',
                              '&:hover': {
                                backgroundColor: 'transparent',
                              },
                              '& .MuiButton-startIcon': {
                                paddingBottom: '1%',
                              },
                            }}
                            variant="outlined"
                            startIcon={<LockIcon />}
                            endIcon={<LaunchIcon />}
                            onClick={viewInCatalogue}
                          >
                            Visualizza sul catalogo
                          </Button>
                        </Grid>
                      }
                    />
                  </Grid>
                ) : (
                  <Grid item key={block}>
                    <InformationContainer
                      key={block}
                      label={Labels[block as keyof typeof Labels]}
                      content={
                        ['statusMonitoring', 'statusService'].includes(block) ? (
                          <Chip
                            size={'small'}
                            sx={{ width: '30%' }}
                            label={data[block as keyof IService]}
                            color={
                              ['attivo', 'online'].includes(data[block as keyof IService])
                                ? 'success'
                                : 'error'
                            }
                          />
                        ) : (
                          data[block as keyof IService]
                        )
                      }
                    />
                  </Grid>
                )
              })}
            </Grid>
          </Grid>
        ))}
      </Grid>
    </>
  )
}
