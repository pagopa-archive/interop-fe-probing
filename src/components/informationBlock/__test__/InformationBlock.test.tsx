import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";
import { InformationBlock } from "../InformationBlock";

const serviceData = {
  eService: "Probing test 3",
  regulator: "Comune di Milano",
  version: "7",
  statusMonitoring: "offline",
  statusService: "attivo",
  statusLastDetection: "03/12/2022, ore 13:30",
};

const reloadInfoBlock = (): void => {
  console.log("reload info block");
};

const viewInCatalogue = (): void => {
  console.log("view in catalogue");
};

describe("InformationBlock", () => {
  test("render component", () => {
    const { container } = render(
      <InformationBlock
        data={serviceData}
        reloadInfoBlock={reloadInfoBlock}
        viewInCatalogue={viewInCatalogue}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
