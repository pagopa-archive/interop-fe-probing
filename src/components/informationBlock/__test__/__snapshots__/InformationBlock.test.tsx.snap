// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InformationBlock > render component 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1xllvgy-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                EROGATORE
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                Comune di Milano
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                VERSIONE
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                7
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-r4jj1p-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                SCHEDA E-SERVICE
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <div
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1bo28fr-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="LockIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                        />
                      </svg>
                    </span>
                    Visualizza sul catalogo
                    <span
                      class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="LaunchIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-10omzie-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-18ngi3z-MuiTypography-root"
        >
          In tempo reale
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-101tbq1-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-l5m4w7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RefreshIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </span>
          Ricarica
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                STATO DEL MONITORAGGIO
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <div
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-7z4xqz-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                  >
                    offline
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                STATO DEL'E SERVICE
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <div
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-qu67q9-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                  >
                    attivo
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                STATO ULTIMA RILEVAZIONE
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                03/12/2022, ore 13:30
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
